src/object/cli_obj.c

```diff
@@ -3989,6 +4035,15 @@ obj_comp_cb_internal(struct obj_auxi_args *obj_auxi)
 
 	if (obj_is_enum_opc(obj_auxi->opc))
 		rc = obj_list_comp(obj_auxi, &iter_arg);
+	else if (obj_auxi->opc == DAOS_OBJ_RPC_KEY2ANCHOR) {
+		daos_obj_key2anchor_t *obj_arg = dc_task_get_args(obj_auxi->obj_task);
+
+		if (!obj_arg->anchor->da_sub_anchors || !obj_is_ec(obj_auxi->obj))
+			goto out;
+		rc = obj_auxi_shards_iterate(obj_auxi, k2a_update_sub_anchor_cb, NULL);
```

Comment:
Hmm, seems no different with other key enumeration operations.  I think obj_list_comp() might just work by using obj_list_dkey_cb().

Reply:
better to keep things separate for better readability?